File: Assignment09/Q3.c
/////////////////////////////////////////////////////////////////////////////////
//
//      Include required header files
//
/////////////////////////////////////////////////////////////////////////////////
#include<stdio.h>

/////////////////////////////////////////////////////////////////////////////////
//
//      Function name:  CountRange
//      Input:          integer
//      Output:         integer
//      Description:    Accept number from user and return count of digits
//                      between 3 and 7 (exclusive range condition as per logic)
//      Date:           28/05/2026
//      Author:         Pratik Subhash Mahale
//
/////////////////////////////////////////////////////////////////////////////////
int CountRange(
                int iNo              // Input number
             )
{
    int iDigit = 0;                 // To store digit
    int iCount = 0;                 // To count required digits

    // Convert negative number into positive
    if(iNo < 0)
    {
        iNo = -iNo;
    }

    // Traverse digits
    while(iNo > 0)
    {
        iDigit = iNo % 10;          // Extract digit

        // Check digit in range 3 to 7
        if((iDigit > 3) && (iDigit < 7))
        {
            iCount++;
        }

        iNo = iNo / 10;             // Remove last digit
    }

    return iCount;                  // Return result
}

/////////////////////////////////////////////////////////////////////////////////
//
//      Application to test CountRange function
//
/////////////////////////////////////////////////////////////////////////////////
int main()
{
    int iValue = 0;                 // Input number
    int iRet = 0;                  // Result

    printf("Enter number\n");

    scanf("%d", &iValue);

    // Function call
    iRet = CountRange(iValue);

    // Output display
    printf("%d\n", iRet);

    return 0;                       // Successful termination
}

/////////////////////////////////////////////////////////////////////////////////
//
//      Input:      2395
//      Output:     1
//
//      Input:      1018
//      Output:     0
//
//      Input:      4521
//      Output:     2
//
//      Input:      9922
//      Output:     0
//
/////////////////////////////////////////////////////////////////////////////////

/////////////////////////////////////////////////////////////////////////////////
//
//      Time Complexity Analysis
//
//      Let n be the number of digits in the input number
//
//      The while loop runs once for each digit
//      Each iteration takes constant time O(1)
//
//      Therefore,
//      Time Complexity = O(n)
//
//      Space Complexity = O(1)
//
/////////////////////////////////////////////////////////////////////////////////

File: Assignment21/Q3.java
///////////////////////////////////////////////////////////////////
//
//      File Name : Q3.java
//      Project   : Logic Building Assignment 21
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Class Name : FactorLogic
//      Method Name: displayFactors
//      Input      : Integer
//      Output     : Void
//      Description: Display all factors of
//                   the given number
//      Date       : 08/06/2026
//      Author     : Pratik Subhash Mahale
//
///////////////////////////////////////////////////////////////////

class FactorLogic
{
    void displayFactors(int num)
    {
        int i = 0;      // Loop counter

        // Convert negative number to positive
        if(num < 0)
        {
            num = -num;
        }

        // Display all factors
        System.out.print("Factors are : ");

        for(i = 1; i <= num; i++)
        {
            if(num % i == 0)
            {
                System.out.print(i + " ");
            }
        }
    }
}

///////////////////////////////////////////////////////////////////
//
//      Main Class
//
///////////////////////////////////////////////////////////////////

public class Q3
{
    public static void main(String args[])
    {
        // Create object of FactorLogic class
        FactorLogic obj = new FactorLogic();

        // Function call
        obj.displayFactors(12);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Input  : 12
//
//      Output :
//      Factors are : 1 2 3 4 6 12
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Time Complexity : O(n)
//
//      Space Complexity : O(1)
//
///////////////////////////////////////////////////////////////////

